src/components/dialogs/equipment-deletion-dialog.js

```diff
@@ -71,19 +111,29 @@ const EquipmentDeletionDialog = ({
     const inputForm = useInputForm();
 
     const [equipmentType, setEquipmentType] = useState(
-        editData?.equipmentType ?? defaultEquipmentType
+        equipmentTypes[editData?.equipmentType] ?? defaultEquipmentType
     );
 
+    const [equipmentsFound, setEquipmentsFound] = useState([]);
+
     const [errors, setErrors] = useState(new Map());
 
-    const [searchMatchingEquipments, equipmentsFound] =
-        useSearchMatchingEquipments(
-            studyUuid,
-            currentNodeUuid,
-            true,
-            equipmentType,
-            makeItems
-        );
+    useEffect(() => {
+        setEquipmentsFound([]);
+        if (Array.isArray(equipmentType.fetcher)) {
+            Promise.all(
+                equipmentType.fetcher.map((fetchPromise) =>
+                    fetchPromise(studyUuid, currentNodeUuid)
+                )
+            ).then((vals) => {
+                setEquipmentsFound(vals?.flat().sort(compareById));
```

Comment:
vals may be null ?

Reply:
You're right, `vals` should not be null (removed the `?`). Also added a catch to handle fetching errors.